### ISSUE TITLE
bug(prefer-enum-over-literal-union): optional string literal unions are not flagged

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod#192]
### 🧩 Description

as description

### 🪜 How to reproduce

```ts
import * as z from 'zod';
z.union([z.literal('foo'), z.literal('bar')]).optional();
```

### ✅ Expected behavior

This kind of expression should be reported

### 💻 System info

```shell
Not relevant
```
